newlib/libc/sys/vita/fs.c

```diff
@@ -592,3 +595,71 @@ int fchdir(int fd)
 	}
 	return 0;
 }
+
+int statvfs(const char *__path, struct statvfs *__buf)
+{
+	if (!__buf || !__path)
+	{
+		errno = EFAULT;
+		return -1;
+	}
+
+	char *realpath = __realpath(__path);
+	if (!realpath)
+	{
+		return -1; // errno already set
+	}
+
+	char drive[16] = {0};
+	int d = __get_drive(realpath);
+	if (d > 15) d = 15; // minus zero-termination
+	strncpy(drive, realpath, d);
```

Comment:
what happen if d is 0?

imo, we have to check a returns of `__get_drive` and `sceIoDevctl`

Reply:
Added check for devctl. That'll cover d==0 case too, since devctl will fail with empty path.